sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -93,5 +95,46 @@ private class AssignmentExpressionBannedWordsFinder : CredentialWordsFinderBase<
                 syntaxNode.Left.IsKnownType(KnownType.System_String, semanticModel) &&
                 syntaxNode.Right.IsKind(SyntaxKind.StringLiteralExpression);
         }
+
+        private class StringLiteralBannedWordsFinder : CredentialWordsFinderBase<LiteralExpressionSyntax>
+        {
+            public StringLiteralBannedWordsFinder(DoNotHardcodeCredentialsBase<SyntaxKind> analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(LiteralExpressionSyntax syntaxNode) =>
+                syntaxNode.GetStringValue();
+
+            protected override string GetVariableName(LiteralExpressionSyntax syntaxNode) =>
+                null;
```

Comment:
this seems like an abuse of the base class

does it mean it will trigger on 

```
foo("password");
```
?

Reply:
There's a lot of logic needed from base class to unify behavior. Thanks to using this base class, the result will behave consistently (and not raise) for 
```C#
var x="password";
foo("password");
```

Reply:
ah ok...
could you add a comment specifying why the Variable name is `null` and what this finder actually does?

do you have test cases with just "password"?
```
foo("password");
s += "password";

string property message => "Please enter password.";
foo("password=");
```